spd/core_metrics_and_figs.py

```diff
@@ -0,0 +1,170 @@
+"""Core metrics and figures for SPD experiments.
+
+This file contains the default metrics and visualizations that are logged during SPD optimization.
+These are separate from user-defined metrics/figures to allow for easier comparison and extension.
+"""
+# pyright: reportMissingImports=false
+
+from typing import Any
+
+import torch
+import wandb
+from jaxtyping import Float, Int
+from matplotlib import pyplot as plt
+from torch import Tensor
+from torch.utils.data import DataLoader
+
+from spd.configs import Config
+from spd.losses import calc_ce_losses
+from spd.models.component_model import ComponentModel
+from spd.plotting import (
+    create_embed_ci_sample_table,
+    plot_causal_importance_vals,
+    plot_ci_histograms,
+    plot_mean_component_activation_counts,
+    plot_UV_matrices,
+)
+from spd.utils.component_utils import calc_ci_l_zero, component_activation_statistics
+from spd.utils.general_utils import calc_kl_divergence_lm
+
+try:
+    from spd.user_metrics_and_figs import compute_user_metrics, create_user_figures
+except ImportError:
+    compute_user_metrics = None
+    create_user_figures = None
+
+
+def create_metrics(
+    model: ComponentModel[Any],
```

Comment:
You've used nn.Module in the generic bound down below, do you want to use that by default?